addons/web/static/src/js/views/kanban/kanban_model.js

```diff
@@ -262,8 +276,19 @@ var KanbanModel = BasicModel.extend({
     },
     /**
      * @override
+     * @param {Array[]} options.activeFilter
      */
     reload: function (id, options) {
+        const element = this.localData[id];
+        // Reset the limit before a reload as there is a mechanism
+        // that may change the limit when a read_group is performed.
+        element.limit = this.loadParams.limit;
```

Comment:
there is an issue here: ungrouped kanban view, change the limit, focus and press enter in search view, new limit is lost